src/validation/loader.rb

```diff
@@ -71,12 +71,33 @@ def answer(absolute_path, section)
         validator.data
       end
 
+      # XXX Extract object for loading configure data?
       def configure_data_tree
-        Validation::Configure.new(config, repo_configure_data).tree
+        Validation::Configure.new(config, combined_configure_data).tree
       end
 
-      def repo_configure_data
-        Data.load(FilePath.configure_file)
+      def combined_configure_data
+        Constants::CONFIGURE_SECTIONS.map do |section|
+          [section, all_questions_for_section(section)]
+        end.to_h
+      end
+
+      def all_questions_for_section(section)
+        [
+          repo_configure_questions,
+          *plugin_configure_questions,
+        ].flat_map do |question_group|
```

Comment:
Well why do we need anything? :slightly_smiling_face: Ruby provides us with several ways to do most things and it's up to us to choose the 'best'/most appropriate for the situation. In this case `flat_map` is a standard method to `map` over an `Enumerable` and then `flatten` the result with one method call. I'd argue that my way is slightly better as:

1. it only involves 1 method call rather than 2;

2. it doesn't involve any mutation, which all things being equal is slightly superior to mutating things in place (as that can have surprising effects).

But there's not much in it either way, they're both pretty simple :slightly_smiling_face:.

Reply:
Okay, I guess the thing that threw me is the `flat_map` implies that we have an object that needs to be flattened. Does the method also iterate through the `dependent` questions correctly? If so then it seems appropriate.

However if it is only the result of how the arrays are combined, then it is sightly confusing.

Reply:
So briefly what this is doing is:

1. We get an array containing the question hashes for the repo, and each enabled plugin in turn: `[repo_configure_questions, *plugin_configure_questions]` (this is identical to your use of `concat` just without mutating the array);

2. We map over this to get an array containing arrays for every question in the repo and every enabled plugin for the given configure section;

3. the `flatten` occurs as part of the `flat_map` to turn this array of arrays into 1 array containing e.g. every domain question from the repo and every domain question from every enabled plugin in order.

So a `flatten` will always be needed in some way, or at least some equivalent of this will be, in order to turn these nested arrays into one big array of question hashes; the same would be needed in your `concat` version as well I believe.

Hope this helps :slightly_smiling_face:.